src/app.js

```diff
@@ -1,196 +1,285 @@
+  form.appendChild(inputElement);
+
+  const addButton = createBtn("add", "Add");
+  form.appendChild(addButton);
+
+  const filterOption = createFilterElement();
+  form.appendChild(filterOption);
+
+  rootElement.appendChild(form);
+
+  const listHeader = createHeading("h2", "list-header", "To Do List for");
+  rootElement.appendChild(listHeader);
+
+  const todoList = createTodoList();
+  rootElement.appendChild(todoList);
+
+  // global variables
+  let userName;
+  let todoItems = [];
+
+  const storageKey = () => `${userName}Todos`;
+
+  const getStoredTodos = () => {
+    if (localStorage.getItem(storageKey()) === null) {
+      return [];
+    }
+    return JSON.parse(localStorage.getItem(storageKey()));
+  };
+
+  const persist = (todoItem) => {
+    localStorage.setItem(storageKey(), JSON.stringify(todoItem));
+  };
+
+  const remove = (todoItem) => {
+    todoItems = todoItems.filter(({ id }) => todoItem.id !== id);
+  };
+
+  const update = (todoItem, todoEl) => {
+    todoItem.text = todoEl.childNodes[0].innerHTML;
+    todoItem.isComplete = todoEl.classList.contains("completed");
+  };
+
+  const createTodoEl = (id) => {
+    const li = document.createElement("li");
+    li.classList.add("todo");
+    li.setAttribute("id", id);
+    return li;
+  };
+
+  const createTextEl = (text) => {
+    const p = document.createElement("p");
+    p.classList.add("todo-text");
+    p.innerHTML = text;
+    return p;
+  };
+
+  const createCompleteButton = (todoEl, editButton, saveTodoItem) => {
+    const button = document.createElement("button");
+    button.innerHTML = "Mark";
+    button.classList.add("complete");
+    button.addEventListener("click", () => {
+      todoEl.classList.toggle("completed");
+      editButton.disabled = todoEl.classList.contains("completed");
       saveTodoItem();
-      button.innerHTML = "Edit";
-      button.classList.remove("save");
-      button.classList.add("edit");
+    });
+    return button;
+  };
+
+  const createCommandButton = (textEl, saveTodoItem) => {
+    const button = document.createElement("button");
+    button.innerHTML = "Edit";
+    button.classList.add("edit");
+    button.addEventListener("click", () => {
+      if (button.innerHTML === "Edit") {
+        textEl.contentEditable = true;
+        textEl.focus();
+        button.innerHTML = "Save";
+        button.classList.remove("edit");
+        button.classList.add("save");
+      } else {
+        textEl.contentEditable = false;
+        saveTodoItem();
+        button.innerHTML = "Edit";
+        button.classList.remove("save");
+        button.classList.add("edit");
+      }
+    });
+    return button;
+  };
+
+  const createDeleteButton = (todoEl, removeTodoItem) => {
+    const button = document.createElement("button");
+    button.innerHTML = "Delete";
+    button.classList.add("delete");
+    button.addEventListener("click", () => {
+      removeTodoItem();
+      todoEl.remove();
+    });
+    return button;
+  };
```

Comment:
try to move all `createXXXX` functions out from `createApp`, they should still work unless they depend on something defined in the `createApp`'s scope